drivers/usb/device/usb_dc_stm32.c

```diff
@@ -488,16 +488,29 @@ int usb_dc_ep_write(const u8_t ep, const u8_t *const data,
 	return ret;
 }
 
-int usb_dc_ep_read(const u8_t ep, u8_t *const data, const u32_t max_data_len,
-		   u32_t * const read_bytes)
+int usb_dc_ep_read_wait(u8_t ep, u8_t *data, u32_t max_data_len,
+			u32_t *read_bytes)
 {
 	struct usb_dc_stm32_ep_state *ep_state = usb_dc_stm32_get_ep_state(ep);
 	u32_t read_count = ep_state->read_count;
 
 	SYS_LOG_DBG("ep 0x%02x, %u bytes, %u+%u, %p", ep,
 		    max_data_len, ep_state->read_offset, read_count, data);
 
-	if (!max_data_len) {
+	if (!EP_IS_OUT(ep)) { /* check if OUT ep */
+		SYS_LOG_ERR("Wrong endpoint direction: 0x%02x", ep);
+		return -EINVAL;
+	}
+
+	if (!data && max_data_len) {
+		SYS_LOG_ERR("Wrong arguments");
+		return -EINVAL;
+	}
+
+	if (!data && !max_data_len) {
```

Comment:
Could we refactor the code to be more clear. E.g.:
```
if (!EP_IS_OUT(ep)) { ... }

if (!max_data_len) { ... }

...

if (data && read_count) {...}
```

Reply:
seems there is no differences

Reply:
We have one less if condition and the code is easier to read at least for me.